website/moneybirdsynchronization/services.py

```diff
@@ -0,0 +1,297 @@
+                    "reference": f"Card payments {datetime.date.today()}",
+                    "financial_mutations_attributes": financial_mutations_attributes
+                    }
+                }
+            )
+
+            for x in range(len(new_cash_payments)):
+                instance = new_cash_payments[x]
+                instance.moneybird_financial_statement_id = statement_response["id"]
+                instance.moneybird_financial_mutation_id = statement_response["financial_mutations"][x]["id"]
+                instance.save()
+
+                mutation_response = self.api.patch(f"financial_mutations/{instance.moneybird_financial_mutation_id}/link_booking",{
+                        "booking_type": "ExternalSalesInvoice",
+                        "booking_id": instance.moneybird_invoice_id,
+                        "price": str(instance.amount),
+                        "description": instance.topic,
+                    })
+
+
+def push_thaliapay_batch(instance):
+    apiservice = MoneybirdAPIService()
+    payments = Payment.objects.filter(batch=instance)
+    tpay_account_id = apiservice.get_financial_account_id("ThaliaPay")
+    apiservice.link_transaction_to_financial_account(tpay_account_id, payments)
+
+
+def update_contact(member):
+    apiservice = MoneybirdAPIService()
+    apiservice.update_contact(Contact.objects.get_or_create(member=member)[0])
+
+
+def delete_contact(instance):
+    apiservice = MoneybirdAPIService()
+    member = Member.objects.get(profile=instance)
+    contact = Contact.objects.get(member=member)
+    apiservice.api.delete(f"contacts/{contact.moneybird_id}")
+    contact.delete()
+
+
+def register_event_registration_payment(instance):
+    apiservice = MoneybirdAPIService()
+    contact_id = apiservice.api.get(f"contacts/customer_id/{settings.MONEYBIRD_UNKOWN_PAYER_ID}")["id"]
+    if instance.payment.paid_by is not None:
+        try:
+            contact_id = Contact.objects.get(member=instance.member).moneybird_id
+        except:
+            pass
+
+
+    start_date = date(instance.event.start, "Y-m-d")
+    project_name = f"{instance.event.title} [{start_date}]"
+    project_id = apiservice.get_project_id(project_name)
+    
+    invoice_info = {
+        "external_sales_invoice": 
+        {
+            "contact_id": contact_id,
+            "reference": str(instance.payment.id),
+            "date": instance.payment.created_at.strftime("%Y-%m-%d"),
+            "source_url": settings.BASE_URL + instance.payment.get_admin_url(),
+            "currency": "EUR",
+            "prices_are_incl_tax": True,
+            "details_attributes":[
+                {
+                    "description": instance.payment.topic,
+                    "price": str(instance.payment.amount),
+                    "project_id": project_id,
+                },
+            ],
+        }
+    }
+
+    try:
+        response = apiservice.api.post("external_sales_invoices", invoice_info)
```

Comment:
Same as above, implement `MoneybirdAPIService.create_external_sales_invoice` (possibly with a lot of (optional) parameters, rather than taking a random dict for invoice info)